packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -819,11 +868,30 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
     }
 
     private onKeyUp(key: React.KeyboardEvent) {
-        if (key.key === 'Shift') {
-            const change: { cursor?: string, shiftKey: boolean } = { shiftKey: false };
-            if (!this.mouseIsDown) {
-                change.cursor = 'default';
+        if (key.key === 'Shift' || key.key === 'Control') {
+            const change: { cursor: string | undefined, shiftKey: boolean, ctrlKey: boolean } = {
+                cursor: this.state.cursor ?? 'default',
+                shiftKey: this.state.shiftKey ?? false,
+                ctrlKey: this.state.ctrlKey ?? false
+            };
+
+            if (key.key === 'Shift') {
+                change.shiftKey = false;
+                if (key.ctrlKey && !this.isSelecting && !this.isPanning) {
+                    change.cursor = 'grabbing';
+                } else if (!this.mouseIsDown && !this.isSelecting) {
+                    change.cursor = 'default';
+                }
+            } else {
+                change.ctrlKey = false;
+                if (key.shiftKey && this.isSelecting && !this.isPanning) {
```

Comment:
I think this should be !thisIsSelecting?

But anyways should be simpler: if either isSelecting or isPanning is true, then cursor should not change under any condition.

And I think shiftKey or ctrlKey state should be updated based on the key being released and no other condition?

Reply:
Fixed.